File: days/day-02/python/part1.py
#!/usr/bin/env python3

from sys import argv

data = [f.split() for f in open(argv[1]).read().split('\n')]

"""
forward X increases the horizontal position by X units.
down X increases the depth by X units.
up X decreases the depth by X units.
"""

xPos = 0
depth = 0

for move in data:
    dir = int(move[1])
    if move[0] == 'forward':
        xPos += dir
    elif move[0] == 'up':
        depth -= dir
    elif move[0] == 'down':
        depth += dir

print(f"Part 1: {xPos*depth}")

File: 2021/days/day-02/nim/part1.nim
import os,strutils,sequtils,sugar
from strformat import fmt

let data = collect(newSeq):
  for x in paramStr(1).lines:
    x.split.toSeq
    
var
  xPos = 0
  depth = 0

for move in data:
  var dir = move[1].parseInt
  case move[0]:
    of "forward":
      xPos += dir
    of "up":
      depth -= dir
    of "down":
      depth += dir
    else:
      echo fmt"Unknown move: {move[0]}"
  
echo fmt"Part 1: {xPos*depth}"